### ISSUE TITLE
performance problems, loading speed

### ISSUE THREAD

**Issue by emilyhobson** [cityenergyproject/seattle#33]
Current site is unacceptably slow and doesn't load on IE (required for OSE). We can address in greater depth in phase 2 but we need the pages to moderately function for phase 1. 

Trouble areas:

1.  Initial load
1. Metric panes - clicking on a pane generates a response after several seconds 
1. page freezing altogether

Thoughts around causes:
- lots of javascript and css files, color ramps
- large number of points (we'll need to support 100k+ ongoing)
- lack of caching

Suggestions: 
Change functionality of building comparison - since you can't see this section and the map at the same time (on a laptop), initiate load of comparison table upon scroll or button action.

**Comment by almccon** [comment]
Our temporary fix of removing the comparison view (9df25b79fdac7edddb92fe0fcc8b0b44976ceb29) seems to be causing the CARTO layers to stack up on each other, instead of removing and replacing the layer when you choose a different metric.

**Comment by almccon** [comment]
I tried making the building comparison `display: none` as default, but that doesn't speed up the loading. Still takes just as much time as if it were rendering the whole thing. 

So we still have change it to actually delay the build of that table until the user clicks to show it. Proceeding with that plan.

**Comment by almccon** [comment]
I narrowed down the slowness to this part of the code: https://github.com/cityenergyproject/seattle/blob/a11e35c1f54732c49324770eb99112d29e86fed6/src/app/views/building_comparison/building_comparison.js#L36-L55

Or at least that's one cause of the slowness on initial load. Continuing that thread in issue #39. Keeping this issue open in case there are other cases to slowness to address.

**Comment by almccon** [comment]
Loading time is much improved, thanks @sconnelley. Looks like this commit did it: eac0925a034289db242ca33dca472e4f14201f13

**Comment by sconnelley** [comment]
yes, the `toColor` method in `BuildingColorBucketCalculator` was taking ~4ms to complete.  Each building calls that method, so for 3k buildings that equals a lot of time.
